api_app/db/repositories/workspaces.py

```diff
@@ -70,6 +67,24 @@ def create_workspace_item(self, workspace_input: WorkspaceInCreate, auth_info: d
 
         return workspace
 
+    def get_address_space_based_on_size(self, workspace_properties: dict):
+
+        # Default the address space to 'small'. This was the pre-existing default
+        address_space_size = workspace_properties.get("address_space_size", "small")
+
+        #773 allow custom sized networks to be requested
+        if (address_space_size == "custom"):
+            # if address_space isn't provided in the input, generate a new one.
+            # TODO: #772 check that the provided address_space is available in the network.
+            return workspace_properties.get("address_space") or self.get_new_address_space()
```

Comment:
So this means that if the user requested a workspace with a custom address_space (e.g. 1.2.3.4/24) then they also need to supply address_space_size=custom?
If yes, wouldn't it make more sense to switch it around and not care about the address_space_size if an address_space was provided?

Reply:
I think I quite like the explicitness of setting `address_space_size=custom` 🤷

Reply:
The confusion I saw is that we say the size is custom but we expect the user to supply the full space and not the custom size. But it's not that important... we can leave it as is.

Reply:
I like the fact its explicit too. Very clear you are "going off piste".